### ISSUE TITLE
How to override @Deprecated?

### ISSUE THREAD

**Issue by wtell400** [springdoc/springdoc-openapi#747]
I have a situation in which one of my setters has a `@java.lang.Deprecated` annotation on an entity. The field itself does not have this annotation.

In swagger, in my schema/model, this fields ends up with the deprecated: true text. I did not expect that, and would like to avoid that.

```java
public class DeprecatedEntity
{
    @Schema(deprecated = false) // Does not work
    private String myNonDeprecatedField;

    public String getMyNonDeprecatedField()
    {
        return myNonDeprecatedField;
    }

    @Schema(deprecated = false) // Does not work
    @Deprecated
    public DeprecatedEntity setMyNonDeprecatedField(String myNonDeprecatedField)
    {
        this.myNonDeprecatedField = myNonDeprecatedField;
        return this;
    }
}
```

I have tried overriding this with `@Schema(deprecated = false)` on both the setter and the field, but this does not seem to work. I assume that processing `@Deprecated` is done by the springdoc library, although I'm not sure what part is springdoc and what part is swagger.

Using springdoc-openapi 1.4.1. See attached project.
[springdoc-issue-override-deprecated.zip](https://github.com/springdoc/springdoc-openapi/files/4825931/springdoc-issue-override-deprecated.zip)

**Comment by bnasslahsen** [comment]
@wtell400, 

 First solution, you override SchemaPropertyDeprecatingConverter:

```java
@Component
public class DeprecatedPropertyCustomizer extends SchemaPropertyDeprecatingConverter {

	@Override
	public Schema resolve(AnnotatedType type, ModelConverterContext context, Iterator<ModelConverter> chain) {
		if (chain.hasNext()) {
			Schema<?> resolvedSchema = chain.next().resolve(type, context, chain);
			if (type.isSchemaProperty() && containsDeprecatedAnnotation(type.getCtxAnnotations()) && !"myNonDeprecatedField".equals(type.getPropertyName()))
				resolvedSchema.setDeprecated(true);
			return resolvedSchema;
		}
		return null;
	}

}

```

Second solution, you use OpenApiCustomiser:

```java
@Bean
public OpenApiCustomiser sortSchemasAlphabetically() {
	return openApi -> {
		if (openApi.getComponents() != null) {
			Schema schema = openApi.getComponents().getSchemas().get("DeprecatedEntity");
			Map<String, Schema> properties = schema.getProperties();
			Schema myNonDeprecatedFieldSchema = properties.get("myNonDeprecatedField");
			myNonDeprecatedFieldSchema.setDeprecated(false);
		}
	};
}
```

**Comment by wtell400** [comment]
These solutions are based on the class and/or field name. I am trying to avoid duplicating 'magic' names and references to a field in other classes. 

The whole idea of using Swagger and `@Schema` is to keep the documentation close to the class/field. Is there a better way of doing this, for example with `@Schema(deprecated = true)` on the same class/method/field that has `@Deprecated`?

**Comment by bnasslahsen** [comment]
The whole idea of springdoc-openapi is to get your documentation the closest to the code, with minimal code changes. 
If the code contains `@Deprecated`, sprindoc-openapi will consider its schema as Deprecated as well.

Here, you want to declare a field on swagger as non deprecated, even with the java code, the field contains `@Depreacted`!

 Additionally to the previous mentioned solutions, you can use the following property that will be available with the next release v1.4.3:

```properties
springdoc.model-converters.deprecating-converter.enabled=false
```

This property, will disable the @Deprecated resolution from springdoc-openapi. So you can use @Schema(deprecated = ...), wherever you want.
